core/modules/qdisp/JobQuery.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ *  Created on: Sep 3, 2015
+ *      Author: jgates
+ */
+
+#ifndef LSST_QSERV_QDISP_JOBQUERY_H_
+#define LSST_QSERV_QDISP_JOBQUERY_H_
+
+// System headers
+#include <atomic>
+#include <memory>
+#include <mutex>
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qdisp/Executive.h"
```

Comment:
The definition for JobDescription is in that file as well, although I intend to move that to its own file before merging.